psydac/feec/multipatch/plotting_utilities.py

```diff
@@ -0,0 +1,300 @@
+# coding: utf-8
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sympy      import lambdify
+from matplotlib import cm
+from matplotlib import colors
+
+from psydac.utilities.utils    import refine_array_1d
+from psydac.fem.basic          import FemField
+from psydac.feec.pull_push     import push_2d_h1, push_2d_hcurl, push_2d_hdiv, push_2d_l2
+
+from mpi4py import MPI
+#==============================================================================
+# some plotting utilities
+# todo (MCP, april 12): merge get_grid_vals_scalar and get_grid_vals_vector into a single function
```

Comment:
Any hope that we can get this TODO done? 😉

Reply:
this file was written by @campospinto, maybe he can explain to us how he wants to do it

Reply:
absolutely :) it's done now in the branch curl-conga:
https://github.com/pyccel/psydac/blob/36893a3ad58941029df53d3d7513480e464c6749/psydac/feec/multipatch/plotting_utilities.py#L86

Reply:
note that it relies on a utility function for Fem fields:
```
def is_vector_valued(u):
    # small utility function, only tested for FemFields in multi-patch spaces of the 2D grad-curl sequence
    # todo: a proper interface returning the number of components of a general FemField would be nice
    return u.fields[0].space.is_product

``` 
which probably deserves a better 'design'

Reply:
If you think that the unified `get_grid_vals` is more convenient than the older two separate functions, I think it is worth including it in this PR.

Reply:
we can include it in this PR, or we can include it in the next one to merge the conga branch, as long as this version works, but both are fine with me

Reply:
I only tested these functions for the cases mentioned (2D multi-patch spaces in the grad-curl sequence), but yes for these cases the unified function is more convenient

Reply:
you can also have a look at the `plot_field` wrapper function which allows for amplitude, components and vector-fields plots given as FemFields or coefficients arrays, it's quite convenient:

https://github.com/pyccel/psydac/blob/498ac393379b076b052efd3cb8ef249b41d6c33c/psydac/feec/multipatch/plotting_utilities.py#L250

Reply:
done in [137d7f1](https://github.com/pyccel/psydac/pull/160/commits/137d7f143b72d4578a47bf313c05516e50dfbe12)